pkg/server/image_pull.go

```diff
@@ -255,15 +256,21 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 	if err != nil {
 		return nil, imagespec.Descriptor{}, errors.Wrap(err, "parse image reference")
 	}
+	client := &http.Client{}
+	if c.config.Registry.DisableHTTP2Client {
+		client.Transport = &http.Transport{
```

Comment:
Looking more closely at `http.Transport`, there are a bunch of fields that get set to default values, and we don't want to lose those values. We should be able to call `http.DefaultTransport.Clone()`, to get a copy, and then just replace `TLSNextProto` on the clone.

Reply:
Done :)